### PR TITLE
Use Authorization header in integration tests

### PR DESCRIPTION
**Pull request by Bond-009**
instead of X-Emby-Authorization

And ensure the response has a successful status code

### CODE REVIEW

**Review by nielsvanvelzen — APPROVED** [review]

File: Jellyfin.Server.Implementations/Security/AuthorizationContext.cs [hunk 1]
```diff
@@ -219,24 +216,7 @@ await using (dbContext.ConfigureAwait(false))
         /// <summary>
         /// Gets the auth.
         /// </summary>
-        /// <param name="httpReq">The HTTP req.</param>
-        /// <returns>Dictionary{System.StringSystem.String}.</returns>
-        private static Dictionary<string, string>? GetAuthorizationDictionary(HttpContext httpReq)
-        {
-            var auth = httpReq.Request.Headers["X-Emby-Authorization"];
```
Comment (nielsvanvelzen) [inline Jellyfin.Server.Implementations/Security/AuthorizationContext.cs:226]: Removing support for this header will cause issues with at least:

- The C# SDK
- The Kodi client
- Jellycon
- The mopidy client/thing

And potentially more clients that changed the header but didn't release yet and third-party clients not includes in my [search](https://github.com/search?q=org%3Ajellyfin%20X-Emby-Authorization&type=code).

Comment (nielsvanvelzen) [reply Jellyfin.Server.Implementations/Security/AuthorizationContext.cs:226]: CC @mcarlton00 @crobibero
See [this gist](https://gist.github.com/nielsvanvelzen/ea047d9028f676185832e51ffaf12a6f) for documentation on API authorization

Comment (Bond-009) [reply Jellyfin.Server.Implementations/Security/AuthorizationContext.cs:226]: This doesn't remove support for `X-Emby-Authorization`, it just removes this duplicate function.

Comment (nielsvanvelzen) [reply Jellyfin.Server.Implementations/Security/AuthorizationContext.cs:226]: Thanks for clarifying, the diff made it look like the header was completely removed.

**Review by Shadowghost — APPROVED** [review]

File: Jellyfin.Server.Implementations/Security/AuthorizationContext.cs [hunk 2]
```diff
@@ -49,14 +49,13 @@ public async Task<AuthorizationInfo> GetAuthorizationInfo(HttpRequest requestCon
         /// <summary>
         /// Gets the authorization.
         /// </summary>
-        /// <param name="httpReq">The HTTP req.</param>
+        /// <param name="httpContext">The HTTP req.</param>
```
Comment (barronpm) [inline Jellyfin.Server.Implementations/Security/AuthorizationContext.cs:52]:
```suggestion
        /// <param name="httpContext">The HTTP request.</param>
```
for consistency

Comment (Bond-009) [reply Jellyfin.Server.Implementations/Security/AuthorizationContext.cs:52]: The HTTP context in this case ;)

Comment (barronpm) [reply Jellyfin.Server.Implementations/Security/AuthorizationContext.cs:52]: :eyes:

**Review by barronpm — APPROVED** [review]

**Pull request merged**